customer_domains.php

```diff
@@ -386,6 +388,39 @@
 				$domainip = $result_ipandport['ip'];
 				$result = \Froxlor\PhpHelper::htmlentitiesArray($result);
 
+				$pubkey = $result['dkim_pubkey'];
+				if((strlen($pubkey) > 20 && strlen($pubkey) < 400) || (strlen($pubkey) < 20 && ((int)\Froxlor\Settings::Get('dkim.dkim_keylength') == 1024))){
+				$sel_value = 1024;
+				}
+				else if((strlen($pubkey) > 400 && strlen($pubkey) < 600) || (strlen($pubkey) < 20 && ((int)\Froxlor\Settings::Get('dkim.dkim_keylength') == 2048))){
+				$sel_value = 2048;
+				}
+				else if((strlen($pubkey) > 600) || (strlen($pubkey) < 20 && ((int)\Froxlor\Settings::Get('dkim.dkim_keylength') == 4096))){
+				$sel_value = 4096;
+				}
+				$keylengthoptions = \Froxlor\UI\HTML::makeoption('1024 Bit', '1024', $sel_value, true, true);
+				$keylengthoptions .= \Froxlor\UI\HTML::makeoption('2048 Bit', '2048', $sel_value, true, true);
+				$keylengthoptions .= \Froxlor\UI\HTML::makeoption('4096 Bit', '4096', $sel_value, true, true);
+
+				if (strlen($pubkey) > 20){
+				    $idna_convert = new \Froxlor\Idna\IdnaWrapper();
+				    $ndomain = $idna_convert -> encode(html_entity_decode($result['domain']));
+/* Lines are too long and destroy usability of input mask
+				    preg_match('/-----BEGIN PUBLIC KEY-----\n(.+)-----END PUBLIC KEY-----/s', $pubkey, $pubkey);
+				    $pubkey = str_replace("\n", "", $pubkey[1]);
+				    $dnsrec = (string) new \Froxlor\Dns\DnsEntry('dkim_' . $result['dkim_id'] . '._domainkey.' . $ndomain . '.', 'TXT', '"v=DKIM1;k=rsa;p=' .$pubkey);
+*/
+				    $pubkey = substr($pubkey, strpos($pubkey, "\n")+1);
+				    $pubkey = substr($pubkey, 0, strrpos($pubkey, "\n")-5);
+				    $pubkey = substr($pubkey, 0, strrpos($pubkey, "\n"));
+				    $pubkey = str_replace("\r","",$pubkey);
+				    $pubkey = str_replace("\n","\"\n\"",$pubkey);
+				    $dnsrec = 'dkim_' . $result['dkim_id'] . '._domainkey.' . $ndomain . '.' . " 18000 IN TXT (\"v=DKIM1;k=rsa;p=\"\n\"" . $pubkey . "\")";
```

Comment:
Please use `Froxlor\Dns\DnsEntry` here, e.g.

```php
$dnsrec = new \Froxlor\Dns\DnsEntry(
    'dkim_' . $result['dkim_id'] . '._domainkey.' . $ndomain . '.',
    'TXT', 
    '"v=DKIM1;k=rsa;p= . $pubkey . "'
);
```

The `__toString()` method of the object will automatically set the system's default TTL and also handle the content wrapping depending on length of the entry